cpp/src/Slice/Parser.h

```diff
@@ -506,7 +499,7 @@ class Container : public virtual SyntaxTreeBase
     bool checkFileMetaData(const StringList&, const StringList&);
     bool validateConstant(const std::string&, const TypePtr&, SyntaxTreeBasePtr&, const std::string&, bool);
 
-    std::map<std::string, ContainedPtr, CICompare> _introducedMap;
+    std::map<std::string, ContainedPtr> _introducedMap;
```

Comment:
When we call `createX` to construct the entity, we already check for naming conflicts, either it's a redefinition error if they're an exact match, or they only differ in case, but we report errors related to identifier names while creating the entity.

In some cases where two entities do only differ in case, it's useful to report the error, but keep parsing anyways as if they're two different entities, to potentially catch more errors. In order to do this, the underlying containers can't care about case anymore, and should treat 'Foo' and 'foo' as distinct identifiers, even though the parser will know that they're erroneous. We still issue all the same errors for it as we did before, but with this change, parsing can just continue past them more effectively.